### PR TITLE
Support of all lammps dump coordinate style

### PR DESCRIPTION
**Pull request by xfanak**
Corresponding tests have been added now.

### CODE REVIEW

**Comment by amcadmus** [comment]
1. You do NOT need to close open a new PR every time you push commits. 
2. please add the UTs for the cases of "scaled" and "scaled & unfolded"

**Comment by xfanak** [comment]
UTs for "scaled" and "scaled & unfolded" have been added.

File: dpdata/lammps/dump.py [hunk 1]
```diff
@@ -51,44 +51,39 @@ def get_natoms_vec(lines) :
...
-    try:
-        posis = get_posi(lines) - orig
-    except ValueError:
-        fposis = get_posi_frac(lines)
-        posis = fposis @ cell
+    posis = np.array(posis)[:,1:4]
+    if sf:
+        posis = (posis%1.)@cell # convert xsu to xs first
+    else:
+        posis = (posis - orig)%np.linalg.norm(cell,axis=1)
```
Comment (amcadmus) [inline dpdata/lammps/dump.py:86]: `%np.linalg.norm(cell,axis=1)` would not work for triclinic cells.

**Review by amcadmus — CHANGES_REQUESTED** [review]

File: dpdata/lammps/dump.py (hunk 1 above)
Comment (xfanak) [reply dpdata/lammps/dump.py]: Fixed. It is %cell.sum(axis=0) now.

File: dpdata/lammps/dump.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def safe_get_posi(lines,cell,orig=np.zeros(3)) :
     if sf:
         posis = (posis%1.)@cell # convert xsu to xs first
     else:
-        posis = (posis - orig)%np.linalg.norm(cell,axis=1)
+        posis = (posis - orig)%cell.sum(axis=0)
```
Comment (amcadmus) [inline dpdata/lammps/dump.py:86]: The 0th dim of cell is the frame. `cell.sum(axis=0)` sums the cell tensors of all frames. It doesn't make sense to me.

**Review by amcadmus — CHANGES_REQUESTED** [review]

File: dpdata/lammps/dump.py (hunk 2 above)
Comment (xfanak) [reply dpdata/lammps/dump.py:86]: The safe_get_posi function handles one frame each time. When the safe_get_posi function is called from system_data, the cell argument passed to safe_get_posi is returned from dumpbox2box function, which is always a 3*3 array.

Comment (amcadmus) [reply dpdata/lammps/dump.py:86]: I see that cell is of shape [3,3], but`%cell.sum(axis=0)` would not give the right wrapped coordinated in a triclinic cell. 

The correct way would be:
1. convert the physical coordinates to scaled coordinates. Notice that scale cell is a cubic
2. wrap the scaled coordinate by `posis%1.`
3. convert the scaled coordinates back to the physical coordinates.

Comment (xfanak) [reply dpdata/lammps/dump.py]: It seems that [x,y,z]@inv(cell) can accurately scale the physical coordinates, where inv is the inverse operation.

File: dpdata/lammps/dump.py [hunk 3]
```diff
@@ -51,45 +51,38 @@ def get_natoms_vec(lines) :
...
-    try:
-        posis = get_posi(lines) - orig
-    except ValueError:
-        fposis = get_posi_frac(lines)
-        posis = fposis @ cell
-    return posis
+    posis = np.array(posis)[:,1:4]
+    if not sf:
+        posis = (posis-orig)@np.linalg.inv(cell)# Convert to scaled coordinates for unscaled coordinates
+    return (posis%1)@cell # Convert scaled coordinates back to Cartesien coordinates
```
Comment (amcadmus) [inline dpdata/lammps/dump.py:85]: @y1xiaoc Is it the correct behavior for scaled coordinates?

**Review by amcadmus — APPROVED** [review]

**Pull request merged**